src/main/java/com/github/maxopoly/environmentaleffects/datarepresentations/MobConfig.java

```diff
@@ -142,8 +145,10 @@ public MobConfig(String identifier, EntityType type, String name,
 
 	public Location findSpawningLocation(Location loc) {
 		for (int i = 0; i < maximumTries; i++) {
-			int x = loc.getBlockX() + rng.nextInt(range * 2) - range;
-			int z = loc.getBlockZ() + rng.nextInt(range * 2) - range;
+			int raduis = rng.nextInt(range - minimumDistanceFromPlayer) + minimumDistanceFromPlayer;
```

Comment:
Am I correct in assuming that the location given to findSpawningLocation is a player's location? 

If yes, then this code makes some sense. If no, I don't follow.